src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/RecommenderPackageFeed.cs

```diff
@@ -90,7 +90,14 @@ public async Task<SearchResult<IPackageSearchMetadata>> RecommendPackagesAsync(C
             // get recommender service and version info
             if (NuGetRecommender == null)
             {
-                NuGetRecommender = await _nuGetRecommender.GetValueAsync();
+                try
+                {
+                    NuGetRecommender = await _nuGetRecommender.GetValueAsync();
```

Comment:
I think we can call [`GetValueAsync(CancellationToken)`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.asynclazy-1.getvalueasync?view=visualstudiosdk-2019#Microsoft_VisualStudio_Threading_AsyncLazy_1_GetValueAsync_System_Threading_CancellationToken_) overload instead.

Reply:
by can... you mean "we should" right? and we want to do that because that is the general pattern when you call async methods...to continue to chain in the cancellation tokens, right?

Reply:
I added the cancellation token